### ISSUE TITLE
Cannot find child for

### ISSUE THREAD

**Issue by imton** [docker/compose#1304]
Guys,

After upgrading to v1.2 and docker v1.6 I am getting "Cannot find child for XXX" manually forced to fire the up command twice and in the second time it goes up ok.

Any idea?

Thanks

**Comment by thomsch98** [comment]
Hi all,

I had the same problem. The same solution worked for me, too. 
Any official fix in sight?

Thx

**Comment by aanand** [comment]
Can you provide a reproducible case?

**Comment by imton** [comment]
@aanand not yet, totally random

**Comment by gchp** [comment]
I'm also seeing this issue. Seems fairly random, and running a second time works fine.

**Comment by robwithhair** [comment]
I'm experiencing the same issue.  Unable do it repeatably though.

**Comment by Tuuleh** [comment]
I have the same issue. I need to consistently type any docker-compose command twice, and second time always works. This started with an update from Ubuntu 14.10 to 15.04 and happens with Compose 1.1 as well as 1.2 on Docker 1.5 as well as 1.6.2 while working on a Rails project.

**Comment by djmaze** [comment]
Same problem here. It started after switching over to the overlayfs storage driver. It happens when starting single containers with `docker run`, too. So it doesn't seem to be related to docker-compose.

**Comment by fikovnik** [comment]
Same here and similarly it happens kind of randomly. 

Docker version 1.6.2, build 7c8fca2

**Comment by tonyhb** [comment]
Happens with docker/docker using 1.6.2, AUFS, so this isn't necessarily a compose issue.

```
vagrant@docker:~$ docker info
Containers: 1
Images: 25
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 27
 Dirperm1 Supported: false
Execution Driver: native-0.2
Kernel Version: 3.13.0-36-generic
Operating System: Ubuntu 14.04.2 LTS
CPUs: 1
Total Memory: 3.861 GiB
Name: docker.local
ID: TRTV:DEPM:FXLY:EXRJ:NIAT:IA4W:73YW:HLN2:KZ2N:2LRW:FLUY:QU5I
Username: tonyhb
Registry: [https://index.docker.io/v1/]
WARNING: No swap limit support
```

Looking in the codebase now and it seems as though it's a transient GraphDB error.  Will edit with any more info I can find, and any solutions.

**Comment by MrMMorris** [comment]
Ran into this with a fresh install of 1.6.2 and AUFS. Fist container failed with `Cannot find child for /blah`. Second time worked fine.

```
$ docker info
Containers: 0
Images: 5
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 5
 Dirperm1 Supported: true
Execution Driver: native-0.2
Kernel Version: 3.19.0-18-generic
Operating System: Ubuntu 14.04.1 LTS
CPUs: 2
Total Memory: 3.858 GiB
Name: ip-10-0-0-76
ID: U2K3:JGPN:POVH:ZERB:C26E:VFXC:ZNFX:T7XM:JMRJ:WT4M:7WDN:OIS3
WARNING: No swap limit support
```

Out with devicemapper issues in with aufs issues I guess...

**Comment by MrMMorris** [comment]
Seems to be doing it for ever 2nd or 3rd container. No compose so this may need to be moved to docker/docker.

ugh.

**Comment by ajnisbet** [comment]
I'm finding for compose 1.3.0RC1 & docker 1.6.2 that instead of working the 2nd time after a `Cannot find child`, compose crashes:
```
Traceback (most recent call last):
  File "<string>", line 3, in <module>
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.main", line 32, in main
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.docopt_command", line 21, in sys_dispatch
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.command", line 34, in dispatch
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.docopt_command", line 24, in dispatch
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.command", line 66, in perform_command
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.main", line 470, in up
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.project", line 221, in up
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.project", line 261, in _get_convergence_plans
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.service", line 279, in convergence_plan
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.service", line 105, in containers
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.container", line 28, in from_ps
TypeError: coercing to Unicode: need string or buffer, NoneType found
```
1.2 is working for me 2nd time around.

**Comment by tonyhb** [comment]
Gonna investigate this some more with a debug build.  Seems to be an issue with Docker's graph database.

**Comment by anarcher** [comment]
I had the same problem. Happens with docker/docker using 1.6.2, AUFS.

**Comment by brendangibat** [comment]
I'm getting the same stack trace as ajnisbet; I'm usually have to restart the docker service as a temporary solution on 1.3rc2.

**Comment by tonyhb** [comment]
Seems to be to do with the upgrade paths from docker engine ~1.2 to ~1.6.  Need more info to find out what's going on.

**Comment by tarragon** [comment]
I am finding this happening using `-s devicemapper`, so I don't think this is AUFS related.

**Comment by drags** [comment]
Also seeing this issue, not using compose, but there doesn't seem to be an issue in `docker/docker` for this.

Details:
- happened after upgrading from 1.6.2 to 1.7.0
- blew away /var/lib/docker/devicemapper after shutting down 1.6 but before starting 1.7
- switched `--storage-driver` from devicemapper to aufs
- re-pulled all images
- this happens on the first start of each uniquely named container, the second start of that container name works fine

Error from docker daemon:
`ERRO[0371] Handler for POST /containers/{name:.*}/start returned error: Cannot start container f11a456adda4799ed211e1fa25bc7e84a45ee843335cadc19c613d1da6487e73: Cannot find child for /<container-name>`

**Comment by sstarcher** [comment]
I'm having this problem with AUFS
docker 1.7.0 and docker-compose 1.3.1

```
Containers: 12
Images: 154
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 178
 Dirperm1 Supported: false
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.13.0-53-generic
Operating System: Ubuntu 14.04.2 LTS
CPUs: 2
Total Memory: 3.859 GiB
Name: ip-10-0-10-113
ID: 5MNJ:Z4BD:F65D:MFJR:I4KW:QCKA:UEFJ:3647:UMUH:4I7Y:ZIOF:BFXO
WARNING: No swap limit support
```

**Comment by zette** [comment]
Same problem on my side. Using RPi with kernel 4.0.7 + AUFS patches and storage driver aufs. 
First run creates container, but without name. Only second attempt to start it works correctly.

```
docker info
Containers: 6
Images: 3
Storage Driver: aufs
Root Dir: /var/lib/docker/aufs
Backing Filesystem: extfs
Dirs: 15
Dirperm1 Supported: true
Execution Driver: native-0.2
Kernel Version: 4.0.7-v7+
Operating System: Raspbian GNU/Linux 7 (wheezy)
CPUs: 4
Total Memory: 961.6 MiB
Name: malina
 ID: OFVF:OUPI:2WPA:J64C:N22F:NLOL:MUK4:ZFZC:KI4F:4HTD:QBBQ:43HQ
Debug mode (server): true
Debug mode (client): false
Fds: 25
Goroutines: 39
System Time: Wed Jul  1 21:50:55 CEST 2015
EventsListeners: 0
Init SHA1: 9183200cc532e132ff6eac70172636ccc5a33724
Init Path: /usr/lib/docker/dockerinit
Docker Root Dir: /var/lib/docker
```

**Comment by yongfengdu** [comment]
I met the same issue.
Docker 1.7.0 + docker-compose 1.4.0dev + ubuntu 15.04

**Comment by gi4nks** [comment]
Same to me.
Docker 1.7.0 + docker-compose 1.3.2 + ubuntu 15.04

**Comment by demetriusnunes** [comment]
+1 here - Docker 1.7.1 + docker-compose 1.3.3 - Linux Mint 17.2

**Comment by docteurklein** [comment]
exact same stack trace as @sstarcher after a few retries of `up -d`.
Then I upgraded to compose 1.3.3, and the problem disappeared. Not saying it is the solution,
 given what @demetriusnunes just said.

Anyway, [here is the log](https://gist.github.com/7bb4d3631a2d8e6fa678)

**Comment by demetriusnunes** [comment]
After stoping and removing all containers, the problem went away. I guess it was an issue when migrating to the new 1.3 version.

**Comment by psychemedia** [comment]
I got this problem using docker-compose in context of vagrant-docker-compose. After running `vagrant up` I get the child not found, re-running I get:

```
==> default: Running provisioner: docker_compose...
    default: Checking for Docker Compose installation...
    default: Running docker-compose rm...
==> default: stdin: is not a tty
==> default: Traceback (most recent call last):
==> default:   File "<string>", line 3, in <module>
==> default:   File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.main", line 31, in main
==> default:   File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.docopt_command", line 21, in sys_dispatch
==> default:   File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.command", line 34, in dispatch
==> default:   File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.docopt_command", line 24, in dispatch
==> default:   File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.command", line 66, in perform_command
==> default:   File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.main", line 247, in rm
==> default:   File "/code/build/docker-compose/out00-PYZ.pyz/compose.project", line 281, in containers
==> default:   File "/code/build/docker-compose/out00-PYZ.pyz/compose.container", line 28, in from_ps
==> default: TypeError
==> default: : 
==> default: coercing to Unicode: need string or buffer, NoneType found
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

/usr/local/bin/docker-compose -f "/vagrant/docker-compose.yml" rm --force

Stdout from the command:



Stderr from the command:

stdin: is not a tty
Traceback (most recent call last):
  File "<string>", line 3, in <module>
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.main", line 31, in main
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.docopt_command", line 21, in sys_dispatch
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.command", line 34, in dispatch
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.docopt_command", line 24, in dispatch
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.command", line 66, in perform_command
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.main", line 247, in rm
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.project", line 281, in containers
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.container", line 28, in from_ps
TypeError: coercing to Unicode: need string or buffer, NoneType found
```

**Comment by cbegin** [comment]
Same. I'm using Ansible to start a container hosting a Tomcat (JVM) server. Ubuntu 14.04 host with phusion/baseimage in the container. Running on Vagrant for development. No compose, no aufs (I'm using devicemapper).

```
Error response from daemon: Cannot start container d5e2a616d72968cbba7f8e24a64c5e79cbca2c38f075e449e576f3756da1fc99: Cannot find child for /tomcat
```
Docker info...
```
Docker version 1.7.1, build 786b29d

Containers: 3
Images: 48
Storage Driver: devicemapper
 Pool Name: docker-252:0-132951-pool
 Pool Blocksize: 65.54 kB
 Backing Filesystem: extfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 1.315 GB
 Data Space Total: 107.4 GB
 Data Space Available: 2.749 GB
 Metadata Space Used: 3.183 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.144 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.77 (2012-10-15)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.16.0-30-generic
Operating System: Ubuntu 14.04.2 LTS
CPUs: 4
Total Memory: 3.86 GiB
Name: vagrant
ID: CMQL:3F6G:BXLP:APZV:72GT:SAI2:35TW:RQQ6:O2SB:553P:HKUZ:JB4R
WARNING: No swap limit support
```

**Comment by cbegin** [comment]
*UPDATE: Possible Cause/Solution*

In my case, the problem was that I had a rogue step in my Ansible deployment tasks that was restarting the dockerhost without stopping the containers. I don't believe the containers were stopping cleanly or being removed (understandable). The solution for me (that made sense anyway) was to stop the dockerhost from restarting as part of the app deployment. Another thing is that I'm using `docker run --rm` (no -d) via an upstart script. So the auto remove likely wasn't working on a sudden shutdown of the dockerhost.

So you may want to check your scripts/automation/tools to see if anything is restarting the dockerhost or otherwise stopping docker processes uncleanly.  

Hope that helps!

**Comment by jebw** [comment]
I've just experienced this but may have a second issue. 

I did a `sudo restart` and when the machine had rebooted, all the containers and images had disappeared. What's odd is the machine is now using devicemapper not aufs and I'm 90% it had previously been using aufs - I know it set off on 1.5 and devicemapper but I blew away /var/lib/docker when we went to 1.6 and I'm sure it got swapped to aufs at that point.

Prior to running restart I'd cleared out old images with the following command

docker rmi $( docker images | grep '<none>' | tr -s ' ' | cut -d ' ' -f 3)

What I've now found is whilst most containers can be restarted one is complaining about files not existing in the devicemapper folder - I'd be inclined to blame this on devicemapper and ubuntu 14.04 though.

**Comment by jebw** [comment]
Further to the last comment, I don't I'd converted that particular machine to AUFS (it was missing the linux-image-extra packages - so I don't see how it could have been). So I think that was a red herring.

**Comment by jebw** [comment]
I may have a means to reproduce this.

We had a machine which had docker installed prior to the ~1.6 install script changes, when the get.docker.com script was changed to install the linux-extras package on ubuntu - forcing the use of AUFS.

After having problems with devicemapper, I'd followed directions in a bug report and installed linux-image-extras-<kernel-version> BUT not linux-extras-image-virtual. The script at get.docker.com installs both of these. I'd recreated /var/lib/docker and switched to use AUFS.

At some point since the kernel has been updated to a slightly newer version but the machine not rebooted. The corresponding linux-image-extras-<kernel-version> was not installed automatically because the linux-image-extras-virtual package wasn't installed.

When the machine was rebooted, it no longer had AUFS available, and docker changed to devicemapper. This meant all the images and containers disappeared.

On trying to run *docker-compose up* I'd hit this bug, I'm guessing the problem was that somewhere outside of the AUFS driver folder there were references to the old containers which had now disappeared  (because of the switch from the AUFS driver to devicemapper).

**Comment by dimfeld** [comment]
I just saw this after upgrading docker (from 1.5 to 1.8) and also switching from devicemapper to the overlay storage driver. `rm -rf /var/lib/docker` and restarting the daemon fixed it for me. So it may be a matter of some stale data from old storage drivers that is corrected after the first attempt to create a container.

**Comment by Andrei-Pozolotin** [comment]
also seeing random need to invoke ```docker run``` twice with v 1.8.1, especially after service restart

**Comment by batmat** [comment]
+1, had to just execute the same command 3 or 4 times to get it going through all the declared containers (7, with 3 data containers in them).

Adding my conf to help with combinatorials: Docker 1.7.1, Compose 1.4.0, devicemapper (loopback, yeah I know it's wrong, about to migrate to overlay storage driver).

**Comment by sazlin** [comment]
+1, I have to run docker-compose commands twice to get them working. Interestingly, I only started hitting this issue after upgrading my linux kernel to 4.1.7 from 3.whatever Xubuntu 15.04 installs by default.

**Comment by dnephin** [comment]
This is unfortunately not something we can fix from compose. This error is coming from the docker daemon when compose tries to create a container.

Since this is not something we can fix from compose, I'm going to close this issue. If you're still experiencing the issue with the latest docker release, please open an issue on the docker repo (and link it from here so others can follow).

Thanks!
